types/index.esm.d.ts

```diff
@@ -774,7 +774,15 @@ export interface Plugin<O = {}> extends ExtendedPlugin {
 	 * @param {object} options - The plugin options.
 	 * @returns {boolean} `false` to cancel the chart update.
 	 */
-	beforeUpdate?(chart: Chart, args: { mode: UpdateMode }, options: O): boolean | void;
+  beforeUpdate?(chart: Chart, args: { mode: UpdateMode }, options: O): boolean | void;
+  /**
+   * @desc Called during the update process, before any chart elements have been created.
+   * This can be used for data decimation by changing the data array inside a dataset.
+	 * @param {Chart} chart - The chart instance.
+	 * @param {object} args - The call arguments.
+	 * @param {object} options - The plugin options.
+   */
+  beforeElementUpdate?(chart: Chart, args: {}, options: O): void;
```

Comment:
we should probably keep `beforeUpdate` and `afterUpdate` next to each other for consistency with the other before/after methods

Reply:
Good call, I moved it below `afterUpdate`.